secure/json-rpc/src/lib.rs

```diff
@@ -737,3 +683,182 @@ mod test {
+        process_transaction_status_response,
+        test_helpers::{create_test_account_state, create_test_state_with_proof},
+        AccountStateResponse, AccountStateWithProofResponse, Bytes, SubmitTransactionResponse,
+        TransactionView, TransactionViewResponse, VMStatusView,
+    };
+    use ureq::Response;
+
+    #[test]
+    fn test_submit_transaction_fuzzer() {
+        let response = generate_submit_transaction_corpus();
+        fuzz_submit_transaction_response(&response);
+    }
+
+    #[test]
+    fn test_get_account_state_fuzzer() {
+        let response = generate_account_state_corpus();
+        fuzz_account_state_response(&response);
+    }
+
+    #[test]
+    fn test_get_transaction_status_fuzzer() {
+        let response = generate_transaction_status_corpus();
+        fuzz_transaction_status_response(&response);
+    }
+
+    // This generates a genuine response for the submit_transaction() JSON RPC API call.
+    pub fn generate_submit_transaction_corpus() -> Vec<u8> {
+        let response = SubmitTransactionResponse {
+            id: 1,
+            jsonrpc: "2.0".to_string(),
+            result: None,
+        };
+        serde_json::to_vec::<SubmitTransactionResponse>(&response).unwrap()
+    }
+
+    // This generates a genuine response for the get_account_state_with_proof() JSON RPC API call.
+    pub fn generate_account_state_corpus() -> Vec<u8> {
+        // Create test account_state_with_proof
+        let account_state = create_test_account_state();
+        let version_height = 100;
+        let account_state_with_proof = create_test_state_with_proof(&account_state, version_height);
+
+        // Use the test account_state_with_proof to generate a template response
+        let account_state = account_state_with_proof.blob.unwrap();
+        let account_state_blob = Bytes::from(&lcs::to_bytes(&account_state).unwrap());
+        let response = AccountStateWithProofResponse {
+            id: 100,
+            jsonrpc: "2.0".to_string(),
+            result: AccountStateResponse {
+                version: account_state_with_proof.version,
+                blob: Some(account_state_blob),
+            },
+        };
+        serde_json::to_vec::<AccountStateWithProofResponse>(&response).unwrap()
+    }
+
+    // This generates a genuine response for the get_account_transaction() JSON RPC API call.
+    pub fn generate_transaction_status_corpus() -> Vec<u8> {
+        let response = TransactionViewResponse {
+            id: 0,
+            jsonrpc: "".to_string(),
+            result: Some(TransactionView {
+                vm_status: VMStatusView::Executed,
+            }),
+        };
+        serde_json::to_vec::<TransactionViewResponse>(&response).unwrap()
+    }
+
+    // This function fuzzes a submit_transaction() response.
+    pub fn fuzz_submit_transaction_response(data: &[u8]) {
+        if let Ok(submit_tx_response) = String::from_utf8(data.to_vec()) {
```

Comment:
@mimoo, the only downside to this approach is that the 'data' input from the fuzzer must be converted to a string to be inserted into a Response, otherwise the fuzzer fails early. I'm considering making more finer grained tests, but hopefully this is an okay starting point..

Reply:
yeah, you're going to have a LOT of inputs that are just going to get dropped here. This would be fine if the real code really used `String::from_utf8()`, as it could be something we would want to fuzz, but it looks like this isn't the case: https://docs.rs/ureq/1.3.0/src/ureq/response.rs.html#368-384 so you might want to really fuzz this logic instead.